packages/scrollbar/src/bar.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+  <transition name="el-scrollbar-fade">
+    <div
+      v-show="visible"
+      ref="instance"
+      :class="['el-scrollbar__bar', 'is-' + bar.key]"
+      @mousedown="clickTrackHandler"
+    >
+      <div
+        ref="thumb"
+        class="el-scrollbar__thumb"
+        :style="thumbStyle"
+        @mousedown="clickThumbHandler"
+      ></div>
+    </div>
+  </transition>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, inject, onBeforeUnmount, onMounted, ref, Ref } from 'vue'
+import { off, on } from '@element-plus/utils/dom'
+import { BAR_MAP, renderThumbStyle } from './util'
+
+export default defineComponent({
+  name: 'Bar',
+  props: {
+    vertical: Boolean,
+    size: String,
+    move: Number,
+  },
+  setup(props) {
+    const instance = ref(null)
+    const thumb = ref(null)
+    const scrollbar = inject('scrollbar', {} as Ref<Nullable<HTMLElement>>)
+    const wrap = inject('scrollbar-wrap', {} as Ref<Nullable<HTMLElement>>)
+    const bar = computed(() => BAR_MAP[props.vertical ? 'vertical' : 'horizontal'])
+    const barStore = ref({})
+    const cursorDown = ref(null)
+    const visible = ref(false)
+
+    const clickThumbHandler = e => {
+      // prevent click event of middle and right button
+      e.stopPropagation()
+      if (e.ctrlKey || [1, 2].includes(e.button)) {
+        return
+      }
+      startDrag(e)
+      barStore.value[bar.value.axis] = (e.currentTarget[bar.value.offset] - (e[bar.value.client] - e.currentTarget.getBoundingClientRect()[bar.value.direction]))
+    }
+
+    const clickTrackHandler = e => {
+      const offset = Math.abs(e.target.getBoundingClientRect()[bar.value.direction] - e[bar.value.client])
+      const thumbHalf = (thumb.value[bar.value.offset] / 2)
+      const thumbPositionPercentage = ((offset - thumbHalf) * 100 / instance.value[bar.value.offset])
+
+      wrap.value[bar.value.scroll] = (thumbPositionPercentage * wrap.value[bar.value.scrollSize] / 100)
+    }
+
+    const startDrag = e => {
+      e.stopImmediatePropagation()
+      cursorDown.value = true
+      on(document, 'mousemove', mouseMoveDocumentHandler)
+      on(document, 'mouseup', mouseUpDocumentHandler)
+      document.onselectstart = () => false
```

Comment:
Shouldn't we store the original `onselectstart` handle and restore it after `mouseup`? We can hijack the event, but we can't just set it to null, what do you think?

Reply:
sounds good

Reply:
You don't need a ref to store this. use a local variable instead.